src/services/prover.rs

```diff
@@ -892,52 +765,178 @@ fn update_requested_proof(
     Ok(())
 }
 
-fn build_sub_proof_request(
-    req_attrs_for_credential: &[RequestedAttributeInfo],
-    req_predicates_for_credential: &[RequestedPredicateInfo],
-) -> Result<SubProofRequest> {
-    trace!("_build_sub_proof_request <<< req_attrs_for_credential: {:?}, req_predicates_for_credential: {:?}",
-           req_attrs_for_credential, req_predicates_for_credential);
+pub(crate) struct CLCredentialProver<'a> {
```

Comment:
Could it be possible to just handle this without the lifetime? Whats the reason we need a ref?

Reply:
It cannot because `CLCredentialProver` holds references requiring explicit lifetime definition